sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/graphpattern/GraphPatterns.java

```diff
@@ -43,6 +48,76 @@ public static TriplePattern tp(RdfSubject subject, RdfPredicate predicate, RdfOb
 		return new TriplesSameSubject(subject, predicate, objects);
 	}
 	
+	public static TriplePattern tp(RdfSubject subject, RdfPredicate predicate, Value... objects) {
+		return tp(subject, predicate, Rdf.objects(objects));
+	}
+	
+	/**
+	 * Create a triple pattern with the given subject, predicate, and object(s)
+	 * 
+	 * @param subject the triple pattern subject
+	 * @param predicate the triple pattern predicate as a {@link IRI}
+	 * @param objects the triples pattern object(s)
+	 * 
+	 * @return a new {@link TriplePattern}
+	 * 
+	 * @see <a
+	 *      href="http://www.w3.org/TR/2013/REC-sparql11-query-20130321/#QSynTriples">
+	 *      Triple pattern syntax</a>
+	 */
+	public static TriplePattern tp(RdfSubject subject, IRI predicate, RdfObject... objects) {
+		return new TriplesSameSubject(subject, Rdf.iri(predicate), objects);
+	}
+	
+	public static TriplePattern tp(RdfSubject subject, IRI predicate, Value... objects) {
+		return tp(subject, Rdf.iri(predicate), Rdf.objects(objects));
+	}
+
+	/**
+	 * Create a triple pattern with the given subject, predicate, and object(s)
+	 * 
+	 * @param subject the triple pattern subject
+	 * @param predicate the triple pattern predicate as a {@link IRI}
+	 * @param objects the triples pattern object(s)
+	 * 
+	 * @return a new {@link TriplePattern}
+	 * 
+	 * @see <a
+	 *      href="http://www.w3.org/TR/2013/REC-sparql11-query-20130321/#QSynTriples">
+	 *      Triple pattern syntax</a>
+	 */
+	public static TriplePattern tp(Resource subject, RdfPredicate predicate, RdfObject... objects) {
+		if (subject instanceof IRI) {
+			return tp(Rdf.iri((IRI)subject), predicate, objects);
+		}
+		return tp(Rdf.bNode(((BNode)subject).getID()), predicate, objects);
+	}
+
+	public static TriplePattern tp(Resource subject, RdfPredicate predicate, Value... objects) {
+		return tp(subject, predicate, Rdf.objects(objects));
+	}
+	
+	/**
+	 * Create a triple pattern with the given subject, predicate, and object(s)
+	 * 
+	 * @param subject the triple pattern subject as a {@link Resource}
+	 * @param predicate the triple pattern predicate as a {@link IRI}
+	 * @param objects the triples pattern object(s)
+	 * 
+	 * @return a new {@link TriplePattern}
+	 * 
+	 * @see <a
+	 *      href="http://www.w3.org/TR/2013/REC-sparql11-query-20130321/#QSynTriples">
+	 *      Triple pattern syntax</a>
+	 */
+	public static TriplePattern tp(Resource subject, IRI predicate, RdfObject... objects) {
```

Comment:
I would prefer more verbose method names. Unless this method existed already.

Reply:
Keep in mind that in a fluent API it kinda makes sense to keep method names short (though you're right that `tp` is _really_ short :)). And yes this was pre-existing. 

As for the style reformatting, sorry about that, not much I can do at this point though - it should have been reformatted when first added to the code base. Redefining our style guide to get our various tools in better sync is on the calendar, but until that has been done this will occasionally happen.